scripts/restore-workspace-protocol.mjs

```diff
@@ -0,0 +1,38 @@
+import fs from 'fs-extra';
+import path from 'node:path';
+import { glob } from 'glob';
+import {execa} from 'execa';
+
+// The following node packages don't belong to the monorepo and must always be references with specific semver.
+const IGNORE_LIST = ["@novu/ntfr-client", "@novu/floating-vue"];
+
+// WIP: Find all package names in the workspace dynamically and eliminate the need for the IGNORE_LIST
+async function getPackageNames() {
+  const { stdout } = await execa`pnpm nx show projects`;
+  return stdout;
+}
+
+
+// Function to update versions of @novu dependencies
+function updateNovuDependencies(dependencies) {
+  for (const [key] of Object.entries(dependencies || {})) {
+    if (key.startsWith('@novu/') && IGNORE_LIST.indexOf(key) === -1) {
+      dependencies[key] = 'workspace:*';
+    }
+  }
+}
+
+function processPackageJson(filePath) {
+  const packageJson = fs.readJsonSync(filePath);
+
+  updateNovuDependencies(packageJson.dependencies);
+  updateNovuDependencies(packageJson.devDependencies);
```

Comment:
we need to cover `peerDependencies` and `optionalDependencies` too, as some packages use them ([ref](https://github.com/novuhq/novu/blob/next/enterprise/packages/billing/package.json#L52))